plugins/esbuild.ts

```diff
@@ -368,23 +365,86 @@ function buildJsxConfig(config?: DenoConfig): BuildOptions | undefined {
   }
 }
 
+function relativePathFromUri(uri: string, basePath?: string): string {
+  if (uri.startsWith("deno:")) {
+    uri = uri.slice("deno:".length);
+  }
+
+  if (uri.startsWith("file://")) {
+    uri = fromFileUrl(uri);
+  }
+
+  return normalizePath(uri, basePath);
+}
+
 function resolveImports(
-  content: string,
+  outfile: OutputFile,
+  basePath: string,
   esm: EsmOptions,
+  metafile: Metafile,
 ): string {
-  return content.replaceAll(
-    /(from\s*)["']([^"']+)["']/g,
-    (_, from, path) => {
-      if (path.startsWith(".") || path.startsWith("/")) {
-        const resolved = path.endsWith(".json")
-          ? path
-          : replaceExtension(path, ".js");
-        return `${from}"${resolved}"`;
-      }
-      const resolved = import.meta.resolve(path);
-      return `${from}"${handleEsm(resolved, esm) || resolved}"`;
-    },
+  let source = outfile.text;
+
+  source = source.replaceAll(
+    /\bfrom\s*["']([^"']+)["']/g,
+    (_, path) =>
+      `from "${resolveImport(path, outfile.path, basePath, esm, metafile)}"`,
   );
+
+  source = source.replaceAll(
+    /\bimport\s*["']([^"']+)["']/g,
+    (_, path) =>
+      `import "${resolveImport(path, outfile.path, basePath, esm, metafile)}"`,
+  );
+
+  source = source.replaceAll(
+    /\bimport\([\s\n]*["']([^"']+)["'](?=[\s\n]*[,)])/g,
+    (_, path) =>
+      `import("${resolveImport(path, outfile.path, basePath, esm, metafile)}"`,
+  );
+
+  return source;
+}
+
+function resolveImport(
+  importPath: string,
+  sourceFile: string,
+  basePath: string,
+  esm: EsmOptions,
+  metafile: Metafile,
+): string {
+  if (importPath.startsWith(".") || importPath.startsWith("/")) {
+    const absoluteImportPath = join(dirname(sourceFile), importPath);
+    const sourcePathOfImport = normalizePath(absoluteImportPath, basePath);
+    const outputOfImport = Object.entries(metafile.outputs)
+      .find(([_, output]) => {
+        if (!output.entryPoint) {
+          return false;
+        }
+
+        const outputRelativeEntryPoint = relativePathFromUri(
+          output.entryPoint,
+          basePath,
+        );
+
+        return outputRelativeEntryPoint === sourcePathOfImport;
+      });
+
+    if (!outputOfImport) {
+      return importPath;
+    }
+
+    const outputPathOfImport = outputOfImport[0];
+    const relativeOutputPathOfImport = relative(
+      dirname(sourceFile),
+      basePath + "/" + outputPathOfImport,
+    );
+
+    return "./" + toUnixPath(relativeOutputPathOfImport);
```

Comment:
Instead of creating this new function, I see two simpler alternatives:

a) Use `normalizePath`:

```js
return "." + normalizePath(relativeOutputPathOfImport);
```

b) Use the `posix` functions [from `std/path`.](https://github.com/lumeland/lume/blob/main/deps/path.ts#L2)

```js
return posix.join("./", relativeOutputPathOfImport);
```

Reply:
I see. Can you try using the posix functions in the logic before this conversion? esmodules are always posix, but I think they are converted to windows in this block:
```js
  const relativeOutputPathOfImport = relative(
    dirname(sourceFile),
    basePath + "/" + outputPathOfImport,
  );
```

Using the posix version from the begining should prevent this conversion:
```js
  const relativeOutputPathOfImport = posix.relative(
    posix.dirname(sourceFile),
    basePath + "/" + outputPathOfImport,
  );
```

Reply:
So I looked into it a bit and the windows path comes from the esbuild `outputFile.path` (`sourceFile` in the `resolveImport` fn). Using the `posix` functions did not help as they only consider `/` a valid path separator. See https://github.com/lumeland/lume/actions/runs/11659711545/job/32460724348?pr=685#step:6:2953

Reply:
Later I'll look into an alternative approach of not using `outputFile.path` but instead the `metafile` paths.

Reply:
Unfortunately I don't have a Windows env to test it with so I had to use your CI - however it now works using the metafile paths

Reply:
No worries. Thanks for your work!